tools/checked-c-convert/ProgramInfo.cpp

```diff
@@ -100,15 +115,26 @@ PointerVariableConstraint::mkString(Constraints::EnvironmentMap &E) {
     ConstAtom *C = E[&VA];
     assert(C != nullptr);
 
+    std::map<uint32_t, Qualification>::iterator q;
     switch (C->getKind()) {
     case Atom::A_Ptr:
+      q = QualMap.find(V);
+      if (q != QualMap.end())
+        if (q->second == ConstQualification)
+          s = s + "const ";
+
       emittedBase = false;
       s = s + "_Ptr<";
 
       caratsToAdd++;
       break;
     case Atom::A_Arr:
     case Atom::A_Wild:
+      q = QualMap.find(V);
```

Comment:
I think that what's intended here is that the `QualMap` is tracking the qualification of the pointers, while the qualification of the pointed values is tracked independently. I convinced myself once that this was the right thing to do but maybe it isn't.